### ISSUE TITLE
RF Slampher pairing a RF, twice press button still no red LED on Slampher

### ISSUE THREAD

**Issue by araczkowski** [arendst/Sonoff-Tasmota#4541]
**Describe the bug**
I'm trying to pair the Slampher (old model before June 2018), according to description in #327
 Unfortunately my devices are only switching very fast and there is no way to go to pairing mode - no red LED flash.
I tried to change SetOption13 but this does not help.

**Does anybody know how to switch the Slampher (old model) to the RF pairing mode on v6.3.0?** 
**Thanks in advance for any tip.**

- [x] _Searched the problem in issues and in the wiki_
- [x] _Hardware used_ : Slampher
- [] _Development/Compiler/Upload tools used_ :
- [x] _If a pre-compiled release or development binary was used, which one?_ : v6.3.0
- [x] _You have tried latest release or development binaries?_ : yes
- [ ] _Provide the output of command_``status 0`` :
```
STATUS 0 OUTPUT HERE - DO NOT DELETE THE MARKERS ABOVE AND BELOW THIS LINE
```

**To Reproduce**
flash Slampher (old model before June 2018) witch v6.3.0
screwed into a socket
quickly press button on Slampher twice
no red LED on Slampher (no pairing mode)

**Expected behavior**
after quickly press button on Slampher twice
the red LED on Slampher should flash one time, then I should be able to pair it

**Screenshots**
no

**Additional context**
no

**(Please, remember to close the issue when the problem has been addressed)**
ok - I will

**Comment by ascillato2** [comment]
Hi,

Please try this alternate pairing method: https://github.com/arendst/Sonoff-Tasmota/wiki/Sonoff-RF#version-2-pairing-instructions

**Comment by araczkowski** [comment]
Hi @ascillato2 thanks for info,

this is  for units produced after 6/2018, my units are from 2017
but of course I tried this method too (many times) ... unfortunately no luck  
no red LED on Slampher  

any other idea?

PS
the red led is fine - I can see the red flash for a moment, on the booting phase, just after the device is connected to power

**Comment by Jason2866** [comment]
Try to upgrade to latest dev version via OTA. Use this http://thehackbox.org/tasmota/sonoff.bin
After that do the command reset 5. This will erase all settings and erase unused flash.
Just wifi settings will retain. Reconfigure your device and try it again.
Strange errors occur with fragements left in flash, so give it a try

**Comment by araczkowski** [comment]
hi @Jason2866 

I did it - unfortunately I have still the same problem on 5 devices - see the details below.

**Maybe the debug log can tell you something?**
**I'm curious if anybody was able to go to the RF pairing mode on Slampher on the Tasmota software?**

![image](https://user-images.githubusercontent.com/1263968/49660083-0696c900-fa46-11e8-8344-2b675549504b.png)

log level 4 (more debug), when I'm quickly pressing button on Slampher twice 
```
17:37:52 SRC: Button
17:37:52 MQT: stat/sonoff/RESULT = {"POWER":"OFF"}
17:37:52 MQT: stat/sonoff/POWER = OFF
17:37:53 APP: Button1 multi-press 1
17:37:53 CFG: Saved to flash at FB, Count 33, Bytes 3584
17:37:54 SRC: Button
17:37:54 MQT: stat/sonoff/RESULT = {"POWER":"ON"}
17:37:54 MQT: stat/sonoff/POWER = ON
17:37:54 APP: Button1 multi-press 1
17:37:54 SRC: Button
17:37:54 MQT: stat/sonoff/RESULT = {"POWER":"OFF"}
17:37:54 MQT: stat/sonoff/POWER = OFF
17:38:00 WIF: Checking connection...
17:38:00 WIF: Connected
17:38:20 WIF: Checking connection...
17:38:20 WIF: Connected
17:38:41 WIF: Checking connection...
17:38:41 WIF: Connected
17:39:00 WIF: Checking connection...
17:39:00 WIF: Connected
17:39:16 APP: Button1 multi-press 1
17:39:16 SRC: Button
17:39:16 MQT: stat/sonoff/RESULT = {"POWER":"ON"}
17:39:16 MQT: stat/sonoff/POWER = ON
17:39:17 CFG: Saved to flash at FA, Count 34, Bytes 3584
17:39:21 WIF: Checking connection...
17:39:21 WIF: Connected
17:39:25 APP: Button1 multi-press 1
17:39:25 APP: Button1 multi-press 2
17:39:26 SRC: Button
17:39:26 MQT: stat/sonoff/RESULT = {"POWER":"OFF"}
17:39:26 MQT: stat/sonoff/POWER = OFF
17:39:26 CFG: Saved to flash at F9, Count 35, Bytes 3584
17:39:33 APP: Button1 multi-press 1
17:39:33 APP: Button1 multi-press 2
17:39:34 SRC: Button
17:39:34 MQT: stat/sonoff/RESULT = {"POWER":"ON"}
17:39:34 MQT: stat/sonoff/POWER = ON
17:39:34 CFG: Saved to flash at F8, Count 36, Bytes 3584
17:39:39 APP: Button1 multi-press 1
17:39:40 APP: Button1 multi-press 2
17:39:40 SRC: Button
17:39:40 MQT: stat/sonoff/RESULT = {"POWER":"OFF"}
17:39:40 MQT: stat/sonoff/POWER = OFF
17:39:40 CFG: Saved to flash at F7, Count 37, Bytes 3584
17:39:40 WIF: Checking connection...
17:39:40 WIF: Connected
17:39:45 APP: Button1 multi-press 1
17:39:45 APP: Button1 multi-press 2
17:39:45 SRC: Button
17:39:45 MQT: stat/sonoff/RESULT = {"POWER":"ON"}
17:39:46 MQT: stat/sonoff/POWER = ON
17:39:46 CFG: Saved to flash at F6, Count 38, Bytes 3584
17:40:00 WIF: Checking connection...
17:40:00 WIF: Connected
```

**Comment by ascillato2** [comment]
@araczkowski 

Hi,

I could not find anything related to slampher RF programming in Tasmota code. So, seems that the button you need to press, should be connected directly to the RF board. Please, check the board.

**Comment by Jason2866** [comment]
The button is on the RF board. It is not under control from the ESP chip / code!

**Comment by araczkowski** [comment]
Hi @ascillato2 Hi @Jason2866 

thanks for checking the code,

I'll check the board and let you know


PS
but according to this: https://github.com/arendst/Sonoff-Tasmota/issues/1513#issuecomment-355516447

> Pairing will be the same as with the original software as it is not serviced by tasmota but by the "other" microcontroller.


and according to this: http://ewelink.coolkit.cc/?p=131
> For the old models that deliver before June 1st, 2018, please learn as below:
Fast press the pairing button on Slampher twice, the red LED blinks once, then press one of the buttons of your RF controller to set pairing for 1-2 second.

I think that I'm doing all correct - I don't believe that original paring needs screwdriver ;)

PS2
I'm able to pare my Sonoff T1 EU on Tasmota in the same way as it is on original software (by pressing and hold the touch button until a beep sound... but this way works only for new units)

**Comment by Jason2866** [comment]
Device could just  be defect...

**Comment by ascillato2** [comment]
Closing this issue as Theo has explained at https://github.com/arendst/Sonoff-Tasmota/issues/1513#issuecomment-355516447 the pairing is not managed by Tasmota.

### Support Information
See [Wiki](https://github.com/arendst/Sonoff-Tasmota/wiki) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.

**Comment by araczkowski** [comment]
Why did you close this unresolved issue?

**Comment by Jason2866** [comment]
Because it cant be related to Tasmota!

**Comment by araczkowski** [comment]
> Because it cant be related to Tasmota!

So you want to announce that Tasmota can  break the RF support and this is fine?

By the @arendst statement
> Pairing will be the same as with the original software...

I understand that the RF should be possible... but in my case (5 devices) it is not
We can try to explain this issue for all the Tasmota users and maybe improve the code 

But closing this without explonation is not helping to make this software better!

**Comment by ascillato** [comment]
> But closing this without explonation is not helping to make this software better!

Please, don't take it personal. We will continue providing help to you as far as we can with your issue.

What it is explained in the other issue (linked in the [closing comment](https://github.com/arendst/Sonoff-Tasmota/issues/4541#issuecomment-445956252)) is that Tasmota don't control the RF pairing. The pairing is done by another MCU (another board) outside the ESP8266 where Tasmota software is. We provide support in the Tasmota Chat and we track Bug reports in github issues. This issue is not a Tasmota Bug due the pairing is managed by another board. _(We need to be organized with issues because of the high amount of them, if we don't manage them properly, they go high very quickly and we can't track them anymore and bug reports are lost. That is why support is done by a very nice and kind group of people at the Tasmota Chat and Bug reports are tracked here)_

So, the pairing is done using the RF Board. In your version device, the pairing is not done by the button as you have test and told here in this thread. The pairing is done by another button or connection that has to be found. 

We want to help you with that so as to add this information to the wiki for other users.

So, please, check the RF board for another button or connections as asked [before](https://github.com/arendst/Sonoff-Tasmota/issues/4541#issuecomment-445918005), and please, provide photos of that so we can help you with that.

**Comment by ascillato** [comment]
@araczkowski 

In Tasmota code there is nothing related to the RF part. So, i'm wondering how the other MCU chip (the RF) is controlling the ESP8266 in your version of SLAMPHER ?

So, we need to do some reverse engineering. Something is missing here.

Could you check with a DMM which GPIOs of the ESP8266 are connected to the RF chip?

As it is now, The button and the led are controlled by the ESP8266. So, how the RF chip sends the toggle to the relay? may be it uses also the button (GPIO0 line) to send the Toggle commands?

**Comment by ascillato2** [comment]
> no offense, I'm glad that you are giving me response but unfortunately it looks like your approach is:
> 
> try new version
> maybe your hardware is broken
> see wiki, close the issue
> c'mon man this is not helping to anybody :(

I don't appreciate your generalization. It is not like that. You should read the last 1000 (thousand issues) addressed in just last 3 months. We are helping A LOT of people in our free time and for free. Complaining is not helping. It is just putting away the technical focus of the issue and it goes off topic, and also is taking away the willing to help you from anyone on the community. If you think that something needs to change or improve in the way everything is managed, github issues is not the place. We have already a channel for that and we are open for discussion: The Tasmota Support Chat (https://discord.gg/Ks2Kzd4). 

I will moderate this thread, hidding the out of topic comments and leave just the technical comments.

**Comment by araczkowski** [comment]
@ascillato2 thanks for reopening 
I'm doing the "investigation" on pcb board 

It looks that the RF receiver is SYN470R, it is connected with EFM8BB10F micro controller (via pin 10) and  the same pin 10 of the EFM8BB10F is also connect to GPIO0 of the ESP8285 - so you right RF chip sends the toggle to the relay via GPIO0 line

I tried to send some RF code even without red light but it doesn't work - looks like I'm not able to put device in the learning mode

I will inform you about my progress with this

**Comment by araczkowski** [comment]
Finally it works :)

and its little bit embarrassing... but it was my mistake

Explanation:
in 2017 when I programed these devices, to get the ESP8285 in flash mode I had to connect GPIO0 to the ground. To connect the button on the Slampher to GPIO0 I moved an 0Ω resistor R21 to R3. So the button was not connected to the RF part. 

Lot of diferent devices were programed by me after this time and I forgot about this "specific hack" for Slampher.

Now I moved this resistor back and all works exectly like it is described :)

Thanks for you help and patience
